prevent warnings when using Puny encoded Hostnames

mimmi20 opened the pull request:
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | yes
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

This PR wants do prevent warings when using Puny encoded hostnames. This is related to #207

mimmi20 committed:
prevent warnings when using Puny encoded Hostnames

Signed-off-by: Thomas Müller <mimmi20@live.de>

Ocramius commented on test/HostnameTest.php:
```diff
@@ -685,4 +685,9 @@ public function testHostnameWithEmptyDomainPart(): void
         self::assertFalse($this->validator->isValid('.com'));
     }
+
+    public function testHostnameWithPunyEncodedDomainPart(): void
+    {
+        self::assertFalse($this->validator->isValid('xn--k.dk'));
+    }
```
Can we also have a case where rhe hostname is valid, yet puny encoded?

mimmi20 replied:
I moved the testcase to an existing data provider. For valid hostnames are also testcases available  https://github.com/laminas/laminas-validator/blob/b4822617887f20a0a82d0c601b813e4f7378cfe4/test/HostnameTest.php#L589

Ocramius replied:
Thanks, indeed we did have them: https://github.com/laminas/laminas-validator/blob/b4822617887f20a0a82d0c601b813e4f7378cfe4/test/HostnameTest.php#L371-L383

mimmi20 replied:
Do you want more tests?

Ocramius replied:
Nope, all good, just need to see why current tests fail :D

mimmi20 replied:
I added also a test for a valid punycoded hostname.

Ocramius commented:
@mimmi20 your last change made the tests go red though? https://github.com/laminas/laminas-validator/actions/runs/7514051558/job/20456503410#step:3:373

mimmi20 committed:
add test with valid punycoded hostname

Signed-off-by: Thomas Müller <mimmi20@live.de>

mimmi20 commented:
> @mimmi20 your last change made the tests go red though? https://github.com/laminas/laminas-validator/actions/runs/7514051558/job/20456503410#step:3:373

I reverted my last change and added the testcases in as seperate tests.

Ocramius reviewed (approved):
:ship: thanks @mimmi20!
